db/DataObject.java

```diff
@@ -16,11 +18,13 @@
 public class DataObject extends DataFactory {
     
     protected final String dataTable = "";
+    protected static String TABLE = "";
     protected int id = 0;
-    protected String uuid;
+    protected UUID uuid;
     
     public DataObject() {
-        this.setUuid(generateUuid());
+        this.setUuid();
+        this.setId(-1);
```

Comment:
`id` is already being set to 0 by default. This means the user doesn't exisist in the table yet. Not sure why it needs to be set to -1?